src/lib/DEVICE/device.cpp

```diff
@@ -119,10 +119,11 @@ static int _devicesUpdate(unsigned long now)
 {
     int32_t core = CURRENT_CORE;
 
-    bool handleEvents = eventFired[core==-1?0:core] || lastConnectionState[core==-1?0:core] != connectionState || lastModelMatch[core==-1?0:core] != connectionHasModelMatch;
+    bool newModelMatch = connectionHasModelMatch && teamraceHasModelMatch;
+    bool handleEvents = eventFired[core==-1?0:core] || lastConnectionState[core==-1?0:core] != connectionState || lastModelMatch[core==-1?0:core] != newModelMatch;
```

Comment:
minor: this really needs some whitespace on those turnery's. Hard AF to read as-is.

Reply:
haha no kidding. Since all of them are the same (there are 6 copies in a row!) I refactored it to just use a variable.